src/TestCase.js

```diff
@@ -14,12 +14,66 @@ const PLATFORM_EXCLUDED_RESOLUTIONS = [
 export const INVALID_REASON_REQUIRED = 'The required dimensions are missing';
 export const INVALID_REASON_RESOLUTION = 'This resolution is not available on this platform';
 
+
+export class TestCasePendingState {
+	failed;
+	description;
+
+	constructor() {
+		this.failed = true;
+		this.description = "Test case is pending"
+	}
+}
+
+export class TestCaseFinishedState {
+	failed;
+	description;
+
+	/**
+	 * @param {string} description
+	 */
+	constructor( description ) {
+		this.failed = false
+		this.description = description
+	}
+}
+
+export class TestCaseIsRunningState {
+	failed;
+	description;
+
+	/**
+	 * @param {string} description
+	 */
+	constructor( description ) {
+		this.failed = true;
```

Comment:
is failed = true correct here?

Reply:
Looking at this state makes me think that maybe we should rename it to "finished", because that's what it's indicating. What do you think?